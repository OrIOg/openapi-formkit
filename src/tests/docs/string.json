{
	"openapi": "3.0.0",
	"info": {
		"title": "Number test",
		"version": "1.0"
	},
	"paths": {
		"/get-example": {
			"get": {
				"operationId": "getExample",
				"summary": "get example data",
				"parameters":  [{
					"in": "query",
					"name": "min",
					"required": true,
					"schema" : {
						"type": "string",
						"minLength": 3
					}
				},{
					"in": "query",
					"name": "min-zero",
					"required": true,
					"schema" : {
						"type": "string",
						"minLength": 0
					}
				}
				,{
					"in": "query",
					"name": "max",
					"required": true,
					"schema" : {
						"type": "string",
						"maxLength": 3
					}
				},{
					"in": "query",
					"name": "minmax",
					"required": true,
					"schema" : {
						"type": "string",
						"minLength": 3,
						"maxLength": 14
					}
				},{
					"in": "query",
					"name": "email",
					"required": true,
					"schema" : {
						"type": "string",
						"format": "email"
					}
				},{
					"in": "query",
					"name": "telephone",
					"required": true,
					"schema" : {
						"type": "string",
						"format": "telephone"
					}
				},{
					"in": "query",
					"name": "url",
					"required": true,
					"schema" : {
						"type": "string",
						"format": "url"
					}
				}],
				"responses": {
					"200": {
						"description": "200 response",
						"content": {
							"application/json": {
								"schema": {
									"$ref" : "#/components/schemas/Example"
								}
							}
						}
					}
				}
			}
		}
	},
	"components": {
		"schemas": {
			"Example": {
				"description": "Example obj",
				"type": "object",
				"properties": {
					"name": {"type": "string"},
					"description": {"type": "string"}
				},
				"required": ["name", "description"]
			}
		}
	}
}