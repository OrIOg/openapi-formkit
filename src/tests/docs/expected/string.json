{
  "/get-example": {
    "get": {
      "$cmp": "FormKit",
      "props": { "type": "form", "method": "get" },
      "children": [
        {
          "$cmp": "FormKit",
          "props": {
            "type": "text",
            "name": "min",
            "label": "min",
            "validation": [["length:3,"], ["required"]]
          }
        },
        {
          "$cmp": "FormKit",
          "props": {
            "type": "text",
            "name": "max",
            "label": "max",
            "validation": [["length:0,3"], ["required"]]
          }
        },
        {
          "$cmp": "FormKit",
          "props": {
            "type": "text",
            "name": "minmax",
            "label": "minmax",
            "validation": [["length:3,14"], ["required"]]
          }
        },
        {
          "$cmp": "FormKit",
          "props": {
            "type": "email",
            "name": "email",
            "label": "email",
            "validation": [["email"], ["required"]]
          }
        },
        {
          "$cmp": "FormKit",
          "props": {
            "type": "tel",
            "name": "telephone",
            "label": "telephone",
            "validation": [["required"]]
          }
        },
        {
          "$cmp": "FormKit",
          "props": {
            "type": "url",
            "name": "url",
            "label": "url",
            "validation": [["url"], ["required"]]
          }
        }
      ]
    }
  }
}
